### PR TITLE
Remove addressbook in java classes from project

### LINKED ISSUE #102: Refactor AddressBook to Client
AddressBook now stores Homerce's clients, let's refactor the code to reflect this.

### PR DESCRIPTION
**Pull request by RuiFengg**
Cleaned up project package structure as well

### CODE REVIEW

File: docs/diagrams/ArchitectureSequenceDiagram.puml [hunk 1]
```diff
@@ -19,7 +19,7 @@ activate model MODEL_COLOR
 model -[MODEL_COLOR]-> logic
 deactivate model
 
-logic -[LOGIC_COLOR]> storage : saveAddressBook(addressBook)
+logic -[LOGIC_COLOR]> storage : saveAddressBook(clientManager)
```
Comment (hanstirtaputra) [inline docs/diagrams/ArchitectureSequenceDiagram.puml:22]: I think we can change the method to saveClientManager

File: src/main/java/seedu/address/MainApp.java [hunk 2]
```diff
@@ -96,24 +96,24 @@ public void init() throws Exception {
      * or an empty address book will be used instead if errors occur when reading {@code storage}'s address book.
      */
     private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
-        Optional<ReadOnlyAddressBook> addressBookOptional;
-        ReadOnlyAddressBook initialData;
+        Optional<ReadOnlyClientManager> addressBookOptional;
```
Comment (khoongwk) [inline src/main/java/seedu/address/MainApp.java:99]: Should this be renamed to clientManagerOptional?

File: src/main/java/seedu/address/model/Model.java [hunk 3]
```diff
@@ -69,22 +70,22 @@
...
 
     /**
      * Sets the user prefs' address book file path.
      */
-    void setAddressBookFilePath(Path addressBookFilePath);
+    void setClientManagerFilePath(Path clientManagerFilePath);
 
     /**
-     * Replaces address book data with the data in {@code addressBook}.
+     * Replaces address book data with the data in {@code clientManager}.
```
Comment (khoongwk) [inline src/main/java/seedu/address/model/Model.java:81]: "address book data" should be changed to "client manager data"

File: src/main/java/seedu/address/model/ModelManager.java [hunk 4]
```diff
@@ -50,35 +52,35 @@
...
-     * Initializes a ModelManager with the given addressBook and userPrefs.
+     * Initializes a ModelManager with the given clientManager and userPrefs.
      */
-    public ModelManager(ReadOnlyUserPrefs userPrefs, ReadOnlyAddressBook addressBook,
+    public ModelManager(ReadOnlyUserPrefs userPrefs, ReadOnlyClientManager clientManager,
                         ReadOnlyServiceManager serviceManager, ReadOnlyRevenueTracker revenueTracker,
                         ReadOnlyExpenseTracker expenseTracker, ReadOnlyAppointmentManager appointmentManager) {
         super();
-        requireAllNonNull(addressBook, userPrefs, serviceManager, revenueTracker, expenseTracker);
+        requireAllNonNull(clientManager, userPrefs, serviceManager, revenueTracker, expenseTracker);
```
Comment (khoongwk) [inline src/main/java/seedu/address/model/ModelManager.java:61]: require appointmentManager to be non-null too

File: src/main/java/seedu/address/model/ModelManager.java [hunk 5]
```diff
@@ -191,7 +193,7 @@ public void setExpenses(List<Expense> expenses) {
 
     /**
      * Returns an unmodifiable view of the list of {@code Expense} backed by the internal list of Expenses
-     * {@code versionedAddressBook}
+     * {@code versionedClientManager}
```
Comment (khoongwk) [inline src/main/java/seedu/address/model/ModelManager.java:196]: This method is for expenses - should "versionedClientManager" be here?

File: src/main/java/seedu/address/model/manager/ReadOnlyClientManager.java [hunk 6]
```diff
@@ -1,12 +1,12 @@
-package seedu.address.model;
+package seedu.address.model.manager;
 
 import javafx.collections.ObservableList;
 import seedu.address.model.client.Client;
 
 /**
  * Unmodifiable view of an address book
```
Comment (khoongwk) [inline src/main/java/seedu/address/model/manager/ReadOnlyClientManager.java:7]: "address book" should be renamed to client manager

File: src/test/java/seedu/address/logic/commands/client/DeleteClientCommandTest.java [hunk 7]
```diff
@@ -76,7 +76,7 @@ public void execute_invalidIndexFilteredList_throwsCommandException() {
 
         Index outOfBoundIndex = INDEX_SECOND_CLIENT;
         // ensures that outOfBoundIndex is still in bounds of address book list
```
Comment (khoongwk) [inline src/test/java/seedu/address/logic/commands/client/DeleteClientCommandTest.java:78]: rename address book to client manager

File: src/test/java/seedu/address/logic/commands/client/EditClientCommandTest.java [hunk 8]
```diff
@@ -122,7 +122,7 @@ public void execute_duplicateClientFilteredList_failure() {
         showClientAtIndex(model, INDEX_FIRST_CLIENT);
 
         // edit client in filtered list into a duplicate in address book
```
Comment (khoongwk) [inline src/test/java/seedu/address/logic/commands/client/EditClientCommandTest.java:124]: Rename address book

File: src/test/java/seedu/address/logic/commands/client/EditClientCommandTest.java [hunk 9]
```diff
@@ -148,7 +148,7 @@ public void execute_invalidClientIndexFilteredList_failure() {
         showClientAtIndex(model, INDEX_FIRST_CLIENT);
         Index outOfBoundIndex = INDEX_SECOND_CLIENT;
         // ensures that outOfBoundIndex is still in bounds of address book list
```
Comment (khoongwk) [inline src/test/java/seedu/address/logic/commands/client/EditClientCommandTest.java:150]: Rename address book

**Review by khoongwk — APPROVED** [review]
LGTM except for a few more javadoc comments which need renaming

**Review by hanstirtaputra — APPROVED** [review]
LGTM

**Pull request merged**